### DISCUSSION TITLE
Questions Regarding VSCode Walkthroughs

### DISCUSSION THREAD

**Discussion by david-pa** [microsoft/vscode-discussions#500] [Extension Development QnA] [answered]
I would like to implement a walkthrough for my extension, and I have been tinkering with the contribution points a bit, but I have a couple of questions:

1. Are walkthroughs _only_ pulled up on the first install of the extension? I know if you register a command to open the walkthrough, the user can access it anytime from the command palette, but will walkthroughs re-appear when the user updates the extension? If not, what methods are there via the VSCode API to present the user with a similar experience when the extension is updated?
2. I created a small walkthrough as a proof of concept, but I noticed that even though I had the `completionEvent` for a step set to `onCommand:`, it was checked off just from clicking on the step (or opening the walkthrough if it is the first step). I saw this behavior when running [the sample](https://github.com/microsoft/vscode-extension-samples/tree/main/getting-started-sample) as well. Here is a snippet so you can see what I'm referencing:
```
{
	"id": "switchToDefault",
	"title": "Going Back to the Text Editor",
	"description": "You are now in the custom editor. To go back to the VSCode text editor, press the 'Switch to Text Editor' button.",
	"media": {
	        "markdown": "media/custom_editor.md"
	},
	"completionEvents": [
		"onCommand:myExtension.showTextEditor"
	]
}
```

**Comment by alefragnani** [comment, marked as answer]
Hi @david-pa ,

> 1. Are walkthroughs only pulled up on the first install of the extension? I know if you register a command to open the walkthrough, the user can access it anytime from the command palette, but will walkthroughs re-appear when the user updates the extension? If not, what methods are there via the VSCode API to present the user with a similar experience when the extension is updated?

You don't have to register a command to open a Walkthrough. This is controlled by VS Code. You just have to _contribute_, and it will be available _forever_. The user can run it as many times as he/she wants, even with every step completed. But, once you update your extension with an updated walkthrough, VS Code detects it (I don't know how) and displays an `updated` tag on it. To sum up, you don't have to worry about it.

> 2. I created a small walkthrough as a proof of concept, but I noticed that even though I had the completionEvent for a step set to onCommand:, it was checked off just from clicking on the step (or opening the walkthrough if it is the first step). I saw this behavior when running [the sample](https://github.com/microsoft/vscode-extension-samples/tree/main/getting-started-sample) as well. Here is a snippet so you can see what I'm referencing:

This seems to be a bug in the current release, or the behavior has changed and the documentation wasn't properly updated. I remember a similar issue (inconsistent behavior) when I was debugging my first walkthrough, a year ago. Back then I was forced to install the extension to properly load the steps completion. I suggest you to open an issue in VS Code repo https://github.com/microsoft/vscode/issues/new/choose to report this issue.

Hope this helps
